### ISSUE TITLE
Subscription List for Google Custom Search as Site Search

### ISSUE THREAD

**Issue by jsamuel** [RequestPolicy/requestpolicy#196]
```
imported trac ticket
created: 2011-02-20 04:42:34
reporter: farp
```

A lot of sites use the google custom search as the search function on their sites (for instance: [http://userscripts.org/search]).
This leads to the following problems with RP:
- If you are browsing a lot of different sites (doing a research, etc.), you have to enable google manually for each sites, which is quite tedious. (Allowing request generally to google is no option IMO.)
- When the request to google for a site is allowed, this may lead to a insecure feeling, since one doesn't know what else besides the search requests are transfered to google.

So, I was wondering if it would be possible to recognize these search queries to google? [[BR]]
If so, this could be used then as an option in RP, which (when enabled) '''automagically''' allows the the google custom (site) search requests, whether request to google are allowed or not.

**Comment by jsamuel** [comment]
```
imported trac comment
created: 2011-02-20 14:03:21
author: eibwen
```

Seems like a duplicate of a combination of #7, at least as understood as / interpreted by [ticket:7#comment:3], and #8.

With full URI regex/wildcard/cidr matching capability, it should be possible to specifically allow the common google search cgi as a destination policy, eg:

{{{
ALLOW /^http:\/\/google.com\/search\?/ from ANY
}}}

This is far too specific for a UI option; however, it should be add-able to the whitelist as above.  Further, this should _NOT_ be "automagic" -- users should have to add their own policy entries to the whitelist.

**Comment by jsamuel** [comment]
```
imported trac comment
created: 2011-02-20 17:58:02
author: justin
```

Someday RP will grow features for path-based restrictions or even site-specific workarounds/options. There will never be completely arbitrary regex matching of every URL, but I consider it reasonably likely to someday add the option for advanced users to define regular expressions to match against paths of specific domains. Anyways, I agree with the sentiment of this ticket in terms of ultimately giving further control over cross-site requests, but it's going to be quite some time before there's a way to do what you're asking for in a very general manner.

In terms of specific, hard-coded optional workarounds for common annoyances encountered using RP, I'll happily entertain any request. However, special case code decreases maintainability and so, at least until RP's more major usability issues are addressed, it's going to take quite a motivating annoyance to justify it, especially given that there are difficult UI questions to be answered for such features even if RP was at the point where they were easy to implement.

**Comment by jsamuel** [comment]
```
imported trac comment
created: 2011-02-23 14:48:09
author: farp
```

Ok, then let me rephrase this:
This shouldn't be a hardcoded option, but rather a subscription list (which should be named like: "Google Custom Site Search" or similar).
This subscription would then contain rules for the above mentioned problem, for example the rule eibwen mentioned.

By subscribing to the list the user should be aware of the (good) outcome and it can be kept up-to-date w/o depending on extension releases.

This Issue depends on #16.

Changed summary.
Added subscription keyword.
